include/deal.II/base/mpi.h

```diff
@@ -371,14 +379,35 @@ namespace Utilities
      * array is the minimum over the i-th entries of the input arrays from each
      * processor.
      *
+     * This function can also be used for ArrayView objects initialized with
+     * data living in CUDA memory if MemorySpace::CUDA is chosen.
+     *
      * Input and output arrays may be the same.
      */
-    template <typename T>
+    template <typename T, typename MemorySpaceType = dealii::MemorySpace::Host>
     void
     min(const ArrayView<const T> &values,
         const MPI_Comm &          mpi_communicator,
         const ArrayView<T> &      minima);
 
+    template <typename T, typename MemorySpaceType = dealii::MemorySpace::Host>
+    void
+    Isend(const ArrayView<const T> &values,
+          MPI_Datatype              datatype,
```

Comment:
Can't you infer the data type from `T`? If that were possible, I think we ought to use this function everywhere because it would finally give us something more generic :-)  (But that would be for a separate patch.)

Reply:
I guess template deduction already works with this version (in case we are using CPU memory) but I need to check.

Reply:
What I meant is that the `MPI_Datatype` seems redundant -- we already have a way to get that from the type `T`.

Reply:
Ah, sure. Let's keep that for a different PR.